### PR TITLE
Changes in fam_options

### PR DESCRIPTION
**Pull request by soumya-pn**

### CODE REVIEW

File: include/fam/fam.h [hunk 1]
```diff
@@ -308,6 +305,9 @@ typedef struct {
     char *numConsumer;
     /** FAM runtime - Default, pmix*/
     char *runtime;
+    char *fam_default_memory_type;
```
Comment (mashoodk) [inline include/fam/fam.h:308]: do add a comment for fam_default_memory_type

File: src/fam-api/fam.cpp [hunk 2]
```diff
@@ -942,6 +933,26 @@ int fam::Impl_::validate_fam_options(Fam_Options *options,
...
+
+    optValueMap->insert({supportedOptionList[IF_DEVICE], famOptions.if_device});
+    if (options && options->fam_default_memory_type)
+        famOptions.fam_default_memory_type = strdup(options->fam_default_memory_type);
+    else if (!config_file_fam_options.empty() &&
+             config_file_fam_options.count("fam_default_memory_type") > 0)
+        famOptions.fam_default_memory_type =
+            strdup(config_file_fam_options["fam_default_memory_type"].c_str());
+    else
+        famOptions.if_device = strdup("");
```
Comment (mashoodk) [inline src/fam-api/fam.cpp:953]: copy paste error ?

**Review by mashoodk — CHANGES_REQUESTED** [review]

File: src/common/fam_options.h [hunk 3]
```diff
@@ -71,6 +69,9 @@ typedef enum {
     RUNTIME,
     /**Number of consumer threads in case of shared memory model**/
     NUM_CONSUMER,
+    FAM_DEFAULT_MEMORY_TYPE,
+    /** Interface device used in the pe side for communication */
```
Comment (mashoodk) [inline src/common/fam_options.h:73]: Interface device used by the pe for communication

**Review by mashoodk — CHANGES_REQUESTED** [review]

**Review by mashoodk — APPROVED** [review]

**Review by faizan-barmawer — APPROVED** [review]
Changes look fine.

Regards,
Faizan

**Pull request merged**